### PR TITLE
Banned users

### PR DESCRIPTION
**Pull request by Kicferk**
This adds a scope for banning users. 

It uses existing permissions page and components to ban users.

I think as banning is an inherently destructive operation. I'm not sure if we should take some sort of snapshot before deleting their memberships, team affiliations and so on. In any case, automatic unbanning would require a bunch more work - like deactivating memberships, and reactivating them if the user is unbanned. Let me know if I should look at it.

Since banning is currently not reversible, I hid actions from the permissions component that displays banned users.
image

One question (and so QQs in the code): when we log the user in, and we detect they are banned, we log them out right after. Should they get a pop up, or get redirected to some sort of page telling them that they have been banned and to contact us if they think it was a mistake or something?

### CODE REVIEW

File: pages/api/scopes/index.ts [hunk 1]
```diff
@@ -36,13 +39,27 @@ export default async function handler(
...
         await prisma.scopes.create({
-          data: {
-            user_uuid: user?.uuid,
-            scope,
-          },
+          data: { user_uuid, scope },
         });
 
+        if (scope === BANNED) {
+          await prisma.users_stripe_products.deleteMany({
```
Comment (declanramsay) [inline pages/api/scopes/index.ts:49]: * Just to speed this up, let's put it in a `Promise.all` so it runs in parallel
* Also will want to update user prop `club_uuid` to null

File: pages/dashboard/index.tsx [hunk 2]
```diff
@@ -36,6 +41,22 @@ const PrismicClubCard = dynamic(() => import('components/prismic/club-card'));
...
   const { data: user, isLoading } = useMe();
+  const { push } = useRouter();
+
+  const userScopes = getPlainScopes(user?.scopes);
+
+  // QQ commonise
+  // QQ Some information to the user that their account has been banned?
+  const handleSignOut = async () => {
+    const data = await signOut({ redirect: false, callbackUrl: '/' });
+    push(data?.url);
```
Comment (declanramsay) [inline pages/dashboard/index.tsx:52]: Let's display a toast, I was looking at introducing these as a general pattern - https://chakra-ui.com/docs/components/toast/usage

**Review by declanramsay — COMMENTED** [review]
Good stuff, I think a few tweaks and we're there :)

File: pages/dashboard/index.tsx (hunk 2 above)
Comment (Kicferk) [reply pages/dashboard/index.tsx:52]: image

**Review by declanramsay — APPROVED** [review]

**Pull request merged**